### DISCUSSION TITLE
Welcome to alba Discussions!

### DISCUSSION THREAD

**Discussion by okuramasafumi** [okuramasafumi/alba#51] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by crazyrunsnail** [comment]
@okuramasafumi Hello! I'm wonder how to  config a default converter from `Time` class to `String`. Thanks!

**Reply by okuramasafumi** [reply]
Hi, there's no "default converter" for a certain class since Alba doesn't know the type of the attribute.
One way to convert time to string is using `attribute` method.

```ruby
class UserResource
  include Alba::Resource
  attribute :created_at do |user|
    user.created_at.strftime('your format here')
  end
end
```

This way you can control how the conversion is done.

Another way is using type feature.

```ruby
class UserResource
  include Alba::Resource
  attributes created_at: [String, true]
end
```

This call `Time#to_s` on `created_at` attribute. Currently you have to give a Proc to modify the conversion method.

If we find it convenient, I'm willing to implement the feature with which we can modify the default converter globally.

```ruby
# This is not implemented yet
Alba.register_type_converter(from: Time, to: String) do |time|
  time.strftime('format')
end
```

**Reply by okuramasafumi** [reply]
https://github.com/okuramasafumi/alba/tree/type-converter
I implemented global default type converter feature. The usage is just calling `Alba.register_type_converter` like above and type feature with `true` as a second argument works with the default converter.
I'm happy to get some feedback!
